Traducir.Web.Net/wwwroot/js/string-search.js

```diff
@@ -0,0 +1,94 @@
+function initializeStringSearch () {
+    const queryDropdowns = document.querySelectorAll("select.js-string-query-filter");
+    const queryTextInputs = document.querySelectorAll("input[type=text].js-string-query-filter");
+    const queryLinks = document.querySelectorAll("a.js-string-query-filter");
+    let initialQueryFilters;
+
+    window.onload = function () {
+        initialQueryFilters = clone(stringQueryFilters);
+
+        setInputsFromCurrentFilters();
+        hookDropdowns();
+        hookTextboxes();
+        hookQuickLinks();
+        hookHistoryPopState();
+    };
+
+    function setInputsFromCurrentFilters() {
+        let i;
+
+        for (i = 0; i < queryDropdowns.length; i++) {
+            const dropdown = queryDropdowns[i];
+            const queryKey = dropdown.getAttribute("data-string-query-key");
+            dropdown.value = stringQueryFilters[queryKey] || 0;
+        }
+
+        for (i = 0; i < queryTextInputs.length; i++) {
+            const textInput = queryTextInputs[i];
+            const queryKey = textInput.getAttribute("data-string-query-key");
+            textInput.value = stringQueryFilters[queryKey];
+        }
+    }
+
+    function hookDropdowns() {
+        for (var i = 0; i < queryDropdowns.length; i++) {
+            queryDropdowns[i].onchange = e => { updateList(e.target, true); };
+        }
+    }
+
+    function hookTextboxes() {
+        var textInputTimeout = null;
+        for (var i = 0; i < queryTextInputs.length; i++) {
+            queryTextInputs[i].onkeyup = function (e) {
+                clearTimeout(textInputTimeout);
+                textInputTimeout = setTimeout(() => { updateList(e.target); }, 500);
+            };
+        };
+    }
+
+    function hookQuickLinks() {
+        for (var i = 0; i < queryLinks.length; i++) {
+            const link = queryLinks.item(i);
+            link.onclick = e => {
+                const queryKey = e.target.getAttribute("data-string-query-key");
+                const queryValue = e.target.getAttribute("data-string-query-value");
+                const dropdown = document.querySelector(`select[data-string-query-key=${queryKey}`);
+                dropdown.value = queryValue;
+                updateList(e.target, true);
+                e.preventDefault();
+            };
+        }
+    }
+
+    function hookHistoryPopState() {
+        window.onpopstate = function (e) {
```

Comment:
I wouldn't mind doing the whole request/response as an HTTP GET instead of via ajax... what do you think?

Reply:
I feel like the page is quite dynamic by using ajax for the filtering and listing part. The alternative is to load the entire page markup every time the user changes one single filter - slower and a waste of bandwidth, isn't it? 🙂 Also the required JS isn't that big or complex.

That doesn't mean that the entire application will be a SPA. For example the users section will be a separate page.

Reply:
let's go ahead with this approach, yeah :)